_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+
+Let's make sure our docker container can run with HTTPS.
+
+First, we need to create a certificate. We can use the .NET CLI for creating self-signed certificates for development. Please note that this is only for development purposes and should not be used in production.
+
+Run the following command to create a self-signed certificate. Use the right command based on the OS you are using and use an appropriate password.
+
+```bash
+# clean existing certs if any
+dotnet dev-certs https --clean
+# create a new cert on macOS/Linux
+dotnet dev-certs https -ep ${HOME}/.aspnet/https/aspnetapp.pfx -p mypass123
+# create a new cert on Windows
+dotnet dev-certs https -ep %USERPROFILE%\.aspnet\https\aspnetapp.pfx -p mypass123
+# trust the cert
+dotnet dev-certs https --trust
+```
+
+Now we can run the container with HTTPS using the below command as we will use the certificate we created above and mount it as a volume.
+
+```bash
+# macOS/Linux
+docker run -it --rm -p 5000:80 -p 5001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=5001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="mypass123" -e ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx -v ${HOME}/.aspnet/https:/https/ okta-chat
+
+# Windows
+docker run -it --rm -p 5000:80 -p 5001:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=5001 -e ASPNETCORE_Kestrel__Certificates__Default__Password="mypass123" -e ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx -v %USERPROFILE%\.aspnet\https:/https/ okta-chat
+```
+
+The app should be accessible at `https://localhost:5001/` now.
+
+In your Okta Developer Portal, go to **Applications** > **Applications** and click the application's name you created using the CLI. Edit the **General Settings** and add URI for **Sign-in redirect URIs** and **Sign-out redirect URIs** as below:
+
+```bash
+Sign-in redirect URI = https://localhost:5001/authorization-code/callback
+Sign-out redirect URI = https://localhost:5001/
+```
+
+Try running the application, and you should see that to open the chat page, you'll be redirected to the Okta Single Sign-On portal and redirected back. You are now successfully authenticated.
+
+### Add Chat Functionality with Vue.js and SignalR
+
+I'll be using Vue for state management since you can take as little or as much as you want. Sometimes it feels like you gotta buy the whole farm just to get a little milk these days when it comes to developing front-end technologies, but I like that with Vue, I can start with a CDN script tag and use it for a single component, a single page, or dive deep with a robust build system on Node. For this exercise, we'll be using a CDN-hosted script.
```

Comment:
```suggestion
We'll use Vue for state management since you can take as little or as much as you want. With Vue, you. can start with a CDN script tag and use it for a single component, a single page, or a deep dive with a robust build system on Node. For this exercise, we'll be using a CDN-hosted script.
```
eliminating folksy colloquialisms that may be unfamiliar/hard to understand for many readers. 
moving toward 2nd person voice ("you") as recommended in style guide